packages/shared/src/hdf5-utils.ts

```diff
@@ -73,14 +67,41 @@ export function cplxType(
   return { class: DTypeClass.Complex, realType, imagType };
 }
 
-export function compoundType(fields: Record<string, DType>): CompoundType {
+export function compoundType<F extends Record<string, DType>>(
+  fields: F,
+): CompoundType<F> {
   return { class: DTypeClass.Compound, fields };
 }
 
-export function printableCompoundType(
-  fields: Record<string, PrintableType>,
-): PrintableCompoundType {
-  return { class: DTypeClass.Compound, fields };
+export const printableCompoundType = compoundType<
+  Record<string, PrintableType>
+>;
```

Comment:
Taking advantage of a quite recent TypeScript syntax here: https://devblogs.microsoft.com/typescript/announcing-typescript-4-7/#instantiation-expressions